tests/test_lazyflow/test_operators/testOpMultiArraySlicer2.py

```diff
@@ -193,3 +193,38 @@ def testWrapped(self):
         for i, slot in enumerate(opSlicer.Slices[1]):
             assert slot.meta.shape == (10, 10, 10, 1)
             assert (slot[...].wait() == 2 * (i + 1)).all()
+
+    def testReshapeDirtyProp(self):
+        class DirtyCountingOp(Operator):
+            Input = InputSlot()
+            Output = OutputSlot()
+
+            def __init__(self, *args, **kwargs):
+                super().__init__(*args, **kwargs)
+                self.invalid_prob_dirty_count = 0
+                self.prob_dirty_count = 0
+
+            def setupOutputs(self):
+                self.Output.meta.assignFrom(self.Input.meta)
+
+            def execute(self, *args, **kwargs):
+                pass
+
+            def propagateDirty(self, inputSlot, subindex, in_roi):
+                self.prob_dirty_count += 1
+                # propagateDirty() roi doesn't match Input slot -
+                # connected operator and own input out of sync
+                # this was encountered when setDirty was called in setupOutputs
+                if not (len(in_roi.start) == len(in_roi.stop) == len(self.Input.meta.shape)):
+                    self.invalid_prob_dirty_count += 1
+
+        test_op = DirtyCountingOp(graph=self.graph)
+        test_op.Input.connect(self.opSlicer.Slices[0])
+        assert test_op.invalid_prob_dirty_count == 0
+        assert test_op.prob_dirty_count == 0
+
+        data = self.opProvider.Input.value
+        reshaped = data.withAxes("tzyxc")
+        self.opProvider.Input.setValue(reshaped)
+        assert test_op.invalid_prob_dirty_count == 0
+        assert test_op.prob_dirty_count == 1
```

Comment:
`invalid_prob_dirty_count` doesn't change anywhere. Should we add another "arrange, act, assert" with Input having a different ndim?

Reply:
it does not change _now_, before it would've. I think that's one of the main points of the test.

Reply:
Ah, OK, now I understand what is being actually tested.